MapboxDirections/MBDirectionOptions.swift

```diff
@@ -0,0 +1,449 @@
+    
+    /**
+     U.S. customary and British imperial units.
+     
+     Distances are measured in miles and feet.
+     */
+    case imperial
+    
+    /**
+     The metric system.
+     
+     Distances are measured in kilometers and meters.
+     */
+    case metric
+    
+    public init?(description: String) {
+        let measurementSystem: MeasurementSystem
+        switch description {
+        case "imperial":
+            measurementSystem = .imperial
+        case "metric":
+            measurementSystem = .metric
+        default:
+            return nil
+        }
+        self.init(rawValue: measurementSystem.rawValue)
+    }
+    
+    public var description: String {
+        switch self {
+        case .imperial:
+            return "imperial"
+        case .metric:
+            return "metric"
+        }
+    }
+}
+
+/**
+ A `RouteShapeFormat` indicates the format of a route’s shape in the raw HTTP response.
+ */
+@objc(MBInstructionFormat)
+public enum InstructionFormat: UInt, CustomStringConvertible {
+    /**
+     The route steps’ instructions are delivered in plain text format.
+     */
+    case text
+    /**
+     The route steps’ instructions are delivered in HTML format.
+     
+     Key phrases are boldfaced.
+     */
+    case html
+    
+    public init?(description: String) {
+        let format: InstructionFormat
+        switch description {
+        case "text":
+            format = .text
+        case "html":
+            format = .html
+        default:
+            return nil
+        }
+        self.init(rawValue: format.rawValue)
+    }
+    
+    public var description: String {
+        switch self {
+        case .text:
+            return "text"
+        case .html:
+            return "html"
+        }
+    }
+}
+
+@objc(MBDirecitonOptions)
+open class DirectionOptions: NSObject, NSSecureCoding, NSCopying {
+    
+    /**
+     Initializes a route options object for routes between the given waypoints and an optional profile identifier.
+     
+     - parameter waypoints: An array of `Waypoint` objects representing locations that the route should visit in chronological order. The array should contain at least two waypoints (the source and destination) and at most 25 waypoints. (Some profiles, such as `MBDirectionsProfileIdentifierAutomobileAvoidingTraffic`, [may have lower limits](https://www.mapbox.com/api-documentation/#directions).)
+     - parameter profileIdentifier: A string specifying the primary mode of transportation for the routes. This parameter, if set, should be set to `MBDirectionsProfileIdentifierAutomobile`, `MBDirectionsProfileIdentifierAutomobileAvoidingTraffic`, `MBDirectionsProfileIdentifierCycling`, or `MBDirectionsProfileIdentifierWalking`. `MBDirectionsProfileIdentifierAutomobile` is used by default.
+     */
+    @objc required public init(waypoints: [Waypoint], profileIdentifier: MBDirectionsProfileIdentifier? = nil) {
```

Comment:
@1ec5 ideally, we don't have to make this change. This would make it so there are no changes to the nav sdk. Noting this is required because of https://github.com/mapbox/MapboxDirections.swift/pull/236/files#diff-c1adc517aeca20c01aad5859dff47b21R207

Are there any ways around this?

Reply:
No, anyways, it’s a good practice for a class with multiple initializers to mark one as `required` (“designated” in Objective-C parlance) and the others as `optional` (convenience).